api/services/search-service.js

```diff
@@ -10,29 +10,37 @@ module.exports = {
         courseSearch: GRIDDY_BASE + "/course?q=",
         courseDataSearch: GRIDDY_BASE + "/course?id="
     },
+
     sendQueryToGriddy: function (endpoint, params, onRequestCompleted) {
         client.get(endpoint + params, function (data) {
             onRequestCompleted(data);
         });
     },
+
     parseCourseSearchResults: function (data) {
+        if (!data || !data.CodeNameMatches){
+            return []
+        }
         var resultsArr = data.CodeNameMatches.split("</li>");
         var matchingCourses = []
         for (var i = 0; i < resultsArr.length - 1; i++) { // last element is always ""
             var courseAsListItem = resultsArr[i];
             var courseCode = courseAsListItem.substring(courseAsListItem.indexOf(">") + 1, courseAsListItem.indexOf(":"));
             var courseFullName = courseAsListItem.substring(courseAsListItem.indexOf(">") + 1);
+            var courseID = courseAsListItem.substring(courseAsListItem.indexOf('=') + 1, courseAsListItem.indexOf('class') - 1);
 
             courseAsListItem = courseAsListItem + "</li>"; // end tag got deleted in the split
             var course = {
-                course: courseCode,
+                courseCode: courseCode,
                 courseFullName: courseFullName,
-                html: courseAsListItem
+                html: courseAsListItem,
```

Comment:
@tash-had do we still need html?

Reply:
@labib-c good point, we don't. I'll remove it